File: 2018 czerwiec/c++/4.cpp
#include <iostream>
#include <fstream>
using namespace std;

int dane1[1005][15], dane2[1005][15];

int main()
{
    ifstream in1("dane1.txt");
    ifstream in2("dane2.txt");
    ofstream out("wynik4_4.txt");

    for (int i = 1; i <= 1000; i++)
    {
        for (int j = 1; j <= 10; j++)
        {
            in1 >> dane1[i][j];
            in2 >> dane2[i][j];
        }
        int a = 1, b = 1;
        while (a <= 10 && b <= 10)
        {
            if (dane1[i][a] <= dane2[i][b])
            {
                out << dane1[i][a] << ' ';
                a++;
            }
            else
            {
                out << dane2[i][b] << ' ';
                b++;
            }
        }

        while (a <= 10)
            out << dane1[i][a++] << ' ';
        
        while (b <= 10)
            out << dane2[i][b++] << ' ';

        out << '\n';
    }    
}

File: 2018 czerwiec/python/4.py
with open("dane1.txt", "r") as f:
    d1 = f.read().strip().split('\n')

with open("dane2.txt", "r") as f:
    d2 = f.read().strip().split('\n')

ans, row = [], 1
with open("wynik4_4.txt", "w") as f:
    for l1, l2 in zip(d1, d2):
        l1 = list(map(int, l1.split()))
        l2 = list(map(int, l2.split()))

        i, j = 0, 0
        while i < len(l1) and j < len(l2):
            if l1[i] <= l2[j]:
                f.write(str(l1[i]) + ' ')
                i += 1
            else:
                f.write(str(l2[j]) + ' ')
                j += 1
        
        while i < len(l1):
            f.write(str(l1[i]) + ' ')
            i += 1
        
        while j < len(l2):
            f.write(str(l2[j]) + ' ')
            j += 1
        f.write('\n')
    